File: sql/run_sql.py
import duckdb
from pathlib import Path

# Path
path = Path(__file__).resolve().parents[1]


# Folder names
sql_folder = path / "sql"


# Get all SQL files
sql_files = sorted(sql_folder.rglob("*.sql"))


# DuckDB execution
# con = duckdb.connect(database=":memory:")
con = duckdb.connect(database="db.db")


for file in sql_files:
    print(f"Running: {file}")
    with open(file, "r", encoding="utf-8") as f:
        sql = f.read()
    con.sql(sql)


# Close connection
con.close()
print("Connection closed")

File: sql/run_sql.R
library(DBI)
library(duckdb)
library(here)

# Path
path <- here()

# Folder names
sql_folder <- file.path(path, "sql")

# Get all SQL files
sql_files <- sort(list.files(
  sql_folder,
  pattern = "\\.sql$",
  full.names = TRUE,
  recursive = TRUE
))


# DuckDB execution
# con <- dbConnect(duckdb::duckdb(), dbdir = ":memory:")
con <- dbConnect(duckdb::duckdb(), dbdir = "db.db")

for (file in sql_files) {
  print(paste("Running:", file))
  sql <- readLines(file, warn = FALSE, encoding = "UTF-8")
  sql <- paste(sql, collapse = "\n")
  dbExecute(con, sql)
}

# Close connection
dbDisconnect(con, shutdown = TRUE)
print("Connection closed")